src/main/java/org/cactoos/iterable/IterableOf.java

```diff
@@ -155,13 +155,13 @@ public boolean equals(final Object other) {
                     () -> Iterable.class.isAssignableFrom(other.getClass()),
                     () -> {
                         final Iterable<?> compared = (Iterable<?>) other;
-                        final Iterator<?> iterator = compared.iterator();
-                        return new Unchecked<>(
-                            new And(
-                                (X input) -> input.equals(iterator.next()),
-                                this
-                            )
-                        ).value();
+                        final Iterator<X> ftr = this.iterator();
+                        final Iterator<?> str = compared.iterator();
+                        boolean failed = false;
+                        while (ftr.hasNext() && str.hasNext() && (!failed)) {
+                            failed = !ftr.next().equals(str.next());
+                        }
+                        return !failed && !ftr.hasNext() && !str.hasNext();
```

Comment:
@iiylll even though this seems technically correct (judging by the test), this is not well readable and maintainable. I propose to add a todo here to transform this code in something more object oriented. I'm not sure how though, but there must be more declarative solutions that that.

Reply:
True, I have another solution with Matched object, but I am not sure if it's correct.
Could I create one more PR for the same issue?

Reply:
@iiylll you can simply modify this one I think. If you desire so, you can close this PR and open a new one, but there is no need IMHO.

Also can you edit your comment to specify who you are addressing in your comment as per Zerocracy policy: http://www.zerocracy.com/policy.html#42